### DISCUSSION TITLE
Using gittool in azure devops

### DISCUSSION THREAD

**Discussion by dnyandeobharambe** [GitTools/GitVersion#3992] [Q&A] [answered]
Let me explain complete scenario and attaching log. I am not getting expected result, I feel either I am doing something fundamentally wrong or trying some scenario which you never tested.
I have existing repos which merge and deploy couple of times and versioning was not exist. Now I am trying to add it based on commands using gittools with gitversion 5.11.1 /5.12.   I am testing my feature branch but irrespective of command it bump major only. I enable debug and I can see command goes fine , but it consider to bump major onlt. In attached log I am using  +semver: minor.
Here is my config (I am ignoring past commit , that is reason ignore exist , and say I want to start version after this date only).
Very first time I provide +semver: patch it bumped to 1.0.0 (was expecting 0.1.1  )

In short I want to ignore all previous merge and start fresh after this date so it can start as above

```yaml
mode: Mainline
major-version-bump-message: '\+semver:\s?(breaking|major)'
minor-version-bump-message: '\+semver:\s?(feature|minor)'
patch-version-bump-message: '\+semver:\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
branches:
  release:
    regex: ^releases?[/-]
    tag: ''
  feature:
    regex: ^feature?[/-]
    tag: ''
```

I appreciate if someone can point the issue or thing I am doing wrong.
Attached log
[log-gittool.txt](https://github.com/GitTools/GitVersion/files/14890462/log-gittool.txt)

**Comment by dnyandeobharambe** [comment]
Overall my feedback is this works fine if I use in fresh new branch. But if I am trying to use it in existing project where previous releases happened then this tool create mess and do not calculate correct version. 
I have to use it in existing project , is there any suggestion?

I have something like this in history  name of branch  release/hawkeye-release-03082024 , is there any way that in config I can set something to ignore such pattern . So it will calculate correct. 

INFO [04/10/24 17:57:39:92] End: Finding branch source of 'release/hawkeye-release-03082024' (Took: 34.18ms)
2024-04-10T17:57:40.3979861Z     INFO [04/10/24 17:57:39:93] Release branch exists -> Version in branch name: 2022023.0.0 with commit count source 5c38f6d90f1a1e7c760f0be8573effe167b62f51
2024-04-10T17:57:40.3980304Z     INFO [04/10/24 17:57:39:93] Release branch exists -> Version in branch name: 3082024.0.0 with commit count source 219c24033977c239d395700c44de0e6501f7508d
2024-04-10T17:57:40.3980977Z     INFO [04/10/24 17:57:39:99] Found multiple base versions which will produce the same SemVer (3082024.1.0), taking oldest source for commit counting (Release branch exists -> Version in branch name)

**Comment by dnyandeobharambe** [comment, marked as answer]
It work with this change

```yaml
mode: Mainline
major-version-bump-message: '\+semver:\s?(breaking|major)'
minor-version-bump-message: '\+semver:\s?(feature|minor)'
patch-version-bump-message: '\+semver:\s?(fix|patch)'
no-bump-message: '\+semver:\s?(none|skip)'
commit-message-incrementing: Enabled
ignore:
  commits-before: 2024-03-09T06:02:00
branches:
  main:
    regex: ^vertest$  # Matches your developement branch
    tag: ''  # No tag associated with the feature branch
    source-branches: []
```
